### DISCUSSION TITLE
Roadmapping: planning for project wrap-up

### DISCUSSION THREAD

**Discussion by robgjansen** [shadow/shadow#2737] [Show and tell]
Things to do to help make shadow easier to maintain over the longer term after the [NSF sponsorship](https://github.com/shadow/shadow/blob/main/docs/nsf_sponsorship.md) ends.

## shadow 3.0 breaking changes
- https://github.com/shadow/shadow/discussions/2496
- revisit stability guarantees themselves. e.g. allow ourselves to drop support for distros that have gone EOL without a major version bump
- Officially drop support for clang (we're already only testing 2 clang builds, to get its diagnostics)
- drop centOS support

## Simplify/reduce CI? (e.g. coverage)
  - Remove coverage from CI since it's not very useful
  - Drop most of the debug build configs. we want release builds everywhere, since optimization-related bugs might only show up on some configs. Running with e.g. debug-assertions in one config should be sufficient though.
  - Stop running CI on push to main. We already require everything passes on PRs before merge, and that PR branch is up to date before merge
  - Jim will take a look at simplifying incremental builds - maybe switch to sccache instead of pre-baked docker images. If sccache breaks can just delete the related steps of the job.
  - We'll drop the docker workflow, and delete the Dockerfile, and delete the documentation page that references the Dockerfile (but keep the one with flags needed to run in Docker). If people start asking for it, tentative plan would be to rework ci/run.sh so that it uses the same Dockerfile(s) that we point them to.

## Documentation

documentation changes we should make, with whoever is most familiar with the topic

- adding/removing platforms from the CI tests (Jim)
- updating the CI container images (Jim)
- what account stores the CI images, and who has access to them (Jim)
- how to configure a new baseline in the benchmark runner (Steve)
- rob-specific instructions for setting up benchmark runner (Rob)
- internal dev documentation for future devs (everyone)
  - rustdoc code documentation (modules, functions, structs, methods, etc)

**Comment by hammer498** [comment]
Out of curiosity, when does the sponsorship end and what happens after that?  The link above says it ends Sept 30 2022

**Reply by robgjansen** [reply]
The sponsorship end date was extended to Sept. 30 2023 (thanks for pointing out the incorrect date, we're updating the docs now to reflect the sponsorship extension). After that, Shadow will return to being an unfunded open-source project with me as lead maintainer, as it was prior to the sponsorship, until/unless we find additional funding streams. We certainly expect the pace of development to slow considerably at that point, but we're hoping that Shadow will be fairly stable for its core function.
